packages/cinterop/src/jvmMain/CMakeLists.txt

```diff
@@ -4,18 +4,40 @@ find_package(JNI)
 set(SWIG_JNI_GENERATED "${CMAKE_SOURCE_DIR}/../../../jni-swig-stub/build/generated/sources/jni")
 set(SWIG_JNI_HELPERS "${CMAKE_SOURCE_DIR}/../../../jni-swig-stub/src/main/jni")
 set(CINTEROP_JNI "${CMAKE_SOURCE_DIR}/../jvm/jni")
+set(REALM_TARGET_LINK_LIBS RealmFFIStatic Realm::ObjectStore)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
+if (CMAKE_SYSTEM_NAME MATCHES "^Windows")
+    MESSAGE("Building JNI for Windows")
+    set(CMAKE_CXX_STANDARD 17)
+    set(CMAKE_CXX_STANDARD_REQUIRED ON)
+    set(REALM_INCLUDE_DIRS "${CMAKE_BINARY_DIR}/src" "${JAVA_INCLUDE_PATH}" "${JAVA_INCLUDE_PATH}/win32" "${CINTEROP_JNI}" "${SWIG_JNI_GENERATED}" "${SWIG_JNI_HELPERS}")
```

Comment:
Wouldn't it be possible to extract the common part of these and only append the platform specifics inside the configuration. This still smells like forgetting to add includes for a single platform.